src/pgen/advection.cpp

```diff
@@ -134,7 +133,7 @@ void ProblemGenerator(MeshBlock *pmb, ParameterInput *pin) {
     auto swarm_d = swarm->GetDeviceContext();
 
     const int gid = pmb->gid;
-    const int max_active_index = swarm->GetMaxActiveIndex();
+    const int max_active_index = new_particle_context.GetNewParticlesMaxIndex();
```

Comment:
```suggestion
    const int max_active_index = new_particles_context.GetNewParticlesMaxIndex();
```
I'm being very picky :)